libgadget/utils/memory.h

```diff
@@ -17,8 +17,8 @@ struct Allocator {
     char name[12];
     Allocator * parent;
 
-    void * rawbase;
-    void * base;
+    char * rawbase;
```

Comment:
I was following the convention of using void* for untyped buffer and temporarily cast to char for offsetting by bytes.

Reply:
I thought of keeping that but we do some arithmetic on the void pointers (in fact that is the whole point of them). Although they have no types we largely use them as pointers to bytes, I think. I was concerned some compiler might do something weird with it if we miss a cast. Is there a benefit to keeping them as void * ?

Reply:
I don't have a strong opinion about enforcing this. No benefits that I can think of other than differentiating when the buffer is actually used for strings (char*) or as a untyped buffer. void * for untyped is a convention followed by most other C libraries (even glibc), so we may run into some frictions, but it's not that bad.

Compilers should trigger an error for direct arithmetic on void * it is undefined behavior offsetting an untyped pointer (we shouldn't be doing that in the first place) But I guess they emit warnings by default because void * was considered char* in very older versions of C.

Reply:
Indeed - and for gcc there is no warning as there is an extension to treat void * as char * for arithmetic.